### ISSUE TITLE
RDA-konversio ruotsinkieliselle tekstiaineistolle

### ISSUE THREAD

**Issue by pknuuti** [KohaSuomi/Koha#1253]
### Tukipyyntö

Vain ruotsinkielinen tekstiaineisto konvertoidaan ruotsinkielisillä konversiosäännöillä. Tietueet, jotka täyttävät ehdot:

000/06 = a
008/35-37 = swe
041a = swe

**Comment by johannaraisa** [comment]
Lisäsin säännöt konversiotyökaluun.

**Comment by johannaraisa** [comment]
Missä kimpassa voin ajaa testin, niin näkee konvertoituuko oikeilla säännöillä?

**Comment by Kati-Helle** [comment]
Vaikkapa Hellessä.

**Comment by johannaraisa** [comment]
> Vaikkapa Hellessä.

Nyt löytyy välivarastosta uudet tiedostot, ja pitäisi olla ajettu ruotsinkielisillä konversiosäännöillä vain tekstiaineisto. Vielä odotellaan Kansalliskirjastolta päivityksiä konversioon, joten jotakin virheitä vielä varmasti löytyy.

**Comment by Kati-Helle** [comment]
Hellen testitietueista huomioita

Ruotsinkielisiä konversiosääntöjä vaikuttaisi käytettäneen myös

Kun 
000/06=t
008/35-37=swe
041a=swe
![kuva](https://github.com/KohaSuomi/Koha/assets/99952449/e3dfb95a-3641-4a82-8bab-0be5830aa601)

Kun 
000/06=i
041d=swe
008/35-37=swe
![kuva](https://github.com/KohaSuomi/Koha/assets/99952449/b7783eb1-cec1-48fe-b0bb-6370d79ece13)

Tämän tietueen 260a-kenttäarvo muodostunut konversiossa suomeksi
![kuva](https://github.com/KohaSuomi/Koha/assets/99952449/ee9c88f3-a715-46b2-bd8d-65fdbba1c4b4)


Kun 
000/06=j
041d=swe
008/35-37=swe
![kuva](https://github.com/KohaSuomi/Koha/assets/99952449/003a70d2-bdc6-495e-902f-eb7662807889)

Huom! 336-338-kenttäarvot suomeksi, kun tietueella 
000/06=a
008/35-37=swe
041a=swe
![kuva](https://github.com/KohaSuomi/Koha/assets/99952449/c0f7bed9-8892-4560-a8f4-a9762b474cb3)

Huom! Osa tietueen 336-338-kenttäarvoista konvertoitunut suomesta ruotsiksi, kun tietueella
000/06=a
008/35-37=swe
041a=swe

![kuva](https://github.com/KohaSuomi/Koha/assets/99952449/54356313-4c93-4d6f-bc99-a9f828b5591f)

![kuva](https://github.com/KohaSuomi/Koha/assets/99952449/2812a5e1-d847-4742-92d5-4d6380e31e48)

**Comment by johannaraisa** [comment]
Ajelin tietueet uudestaan, kun huomasin että kontrollikentät oli hävinneet.

Odotellaan vielä Kansalliskirjastoilta korjauksia. Mutta mitä itse katselin, niin tuossa sv-tiedostossa näytti olevan ainoastaan tekstiaineistot ja muut olivat menneet fi-tiedostoon. 

Käsittääkseni niissä fi-mäppäyksissä yritetään myös tulkita minkä kielinen aineisto on kyseessä ja siksi siellä taitaa tulla ruotsinkielisiäkin termejä. @AnttHeik korjaa, jos olen väärässä?

**Comment by AnttHeik** [comment]
> Käsittääkseni niissä fi-mäppäyksissä yritetään myös tulkita minkä kielinen aineisto on kyseessä ja siksi siellä taitaa tulla ruotsinkielisiäkin termejä. @AnttHeik korjaa, jos olen väärässä?

Joo, näin on. Mielestäni sen pitäisi tulkita 040b-kentässä olevasta swe-arvosta, että nyt käännetään ruotsiksi kaikki mahdollinen tai mitä nyt järkevästi pystyy kääntämään tai avaamaan. Nyt tuolla on kuitenkin sitten käännösvirheitä sekä suomenkielisessä että ruotsinkielisessä konversiossa. Näihin juuri pyysin korjauksia.

**Comment by Kati-Helle** [comment]
Kontrollikentät ovat nyt mukana. Esimerkki yhdestä uudelleen konvertoidusta tietueesta:
![kuva](https://github.com/KohaSuomi/Koha/assets/99952449/95b354a6-0b71-41c6-8c8f-d3e8953ebff2)

**Comment by AnttHeik** [comment]
> Kontrollikentät ovat nyt mukana. Esimerkki yhdestä uudelleen konvertoidusta tietueesta: 

Hmm, ei näytä ymmärtävän kääntää ruotsiksi tuota 260a-kenttää, vaikka nuo oli korjattu, mutta olikohan se niin vain ruotsinkielisessä konversiossa.

**Comment by johannaraisa** [comment]
> > Kontrollikentät ovat nyt mukana. Esimerkki yhdestä uudelleen konvertoidusta tietueesta:
> 
> Hmm, ei näytä ymmärtävän kääntää ruotsiksi tuota 260a-kenttää, vaikka nuo oli korjattu, mutta olikohan se niin vain ruotsinkielisessä konversiossa.

Joo, tuo näyttää olevan sellainen jolle on tehty konversio fi-mäppäyksillä.

**Comment by johannaraisa** [comment]
Näyttää, että noille 336-, 337- ja 338-kentille se yrittää arvata kielen riippumatta mäppäyksestä. Nyt niihin ruotsinkielisiin mäppäyksiin erotellaan vain tekstiaineistot, joten noista muista tulee sekakielisiä.

**Comment by AnttHeik** [comment]
> Näyttää, että noille 336-, 337- ja 338-kentille se yrittää arvata kielen riippumatta mäppäyksestä. Nyt niihin ruotsinkielisiin mäppäyksiin erotellaan vain tekstiaineistot, joten noista muista tulee sekakielisiä.

Jeps, tuo Katin esimerkkitietue näyttää olevan alun perin sekakielinen muutenkin, mutta fi-konversio tajuaa kääntää ruotsin kielelle vain osan kentistä: (336-, 337- ja 338-kentät ja 600c), kun taas pari muuta pysyy suomena (260a ja 700e).

Onko Helle-kimppa muuten ainoa, jossa vanhoilla Finmarcista Marc21:seen konvertoiduilla tietueilla on mukana 040b-kenttä? Ainakaan OUTIssa sitä nimittäin ei ole vanhoilla tietueilla.

**Comment by AnttHeik** [comment]
Voisitko Johanna ajaa näistä TäTin tietueista erilliset sekä ruotsinkielisen että suomenkielisen konversiotiedostoversion, niin näen miten nuo käyttäytyvät?

1853915
1825979
1846867
1846869
1820880
2363984
2263232
2231394
2391423
1813743
1908787
1858442

**Comment by johannaraisa** [comment]
> Voisitko Johanna ajaa näistä TäTin tietueista erilliset sekä ruotsinkielisen että suomenkielisen konversiotiedostoversion, niin näen miten nuo käyttäytyvät?
> 
> 1853915 1825979 1846867 1846869 1820880 2363984 2263232 2231394 2391423 1813743 1908787 1858442

Nyt on nämä ajettu kummallakin mäppäyksellä.

**Comment by johannaraisa** [comment]
Liittyy tikettiin https://github.com/KohaSuomi/Koha/issues/834

**Comment by AnttHeik** [comment]
Jeps, eli ruotsinkielisessä konversiossa on tällä hetkellä ongelmana se, että joissain tapauksissa 336-kentän termit kääntyvät suomen kielelle, vaikka tietueessa olisi 040b-kentässä arvo swe. Vaikuttaisi siltä, että tätä tapahtuu, jos 336-kenttää on toistettu:
![kuva](https://github.com/KohaSuomi/Koha/assets/116169663/0db8e1b3-109c-4831-9cb2-55e8049d100f)

Jos 040b-kenttää ei ole ollenkaan, niin 336-kentässä sama ongelma kaikessa aineistossa:
![kuva](https://github.com/KohaSuomi/Koha/assets/116169663/63d92f7a-9948-4214-85ae-1aafcf7b9211)

Näistä on raportoitu Kansalliskirjastollekin.

**Comment by AnttHeik** [comment]
Mietin vähän noita ISBD-ajan muita ruotsinkielisiä aineistoja, jotka on kuvailtu ruotsin kielellä eli lähinnä äänitteitä ja nuotteja. Mitä Johanna luulet, pystyisikö tätä ruotsinkieliseen mappaykseen mukaan otettavien tietueiden ehtoa muuttamaan alla olevalla tavalla vai olisiko järkevämpi vain lisätä noihin tietueisiin erämuokkauksella 040b-kenttä, jossa kielikoodi swe ja ajaa ne sitten suomisäännöillä?

Eli mukaan tietueet, joissa

000/06 = a
008/35-37 = swe
041a = swe

tai

336a = Musik (notation)
tai
336a = Musik (performance)
tai
336a = Tal

**Comment by johannaraisa** [comment]
> Mietin vähän noita ISBD-ajan muita ruotsinkielisiä aineistoja, jotka on kuvailtu ruotsin kielellä eli lähinnä äänitteitä ja nuotteja. Mitä Johanna luulet, pystyisikö tätä ruotsinkieliseen mappaykseen mukaan otettavien tietueiden ehtoa muuttamaan alla olevalla tavalla vai olisiko järkevämpi vain lisätä noihin tietueisiin erämuokkauksella 040b-kenttä, jossa kielikoodi swe ja ajaa ne sitten suomisäännöillä?
> 
> Eli mukaan tietueet, joissa
> 
> 000/06 = a 008/35-37 = swe 041a = swe
> 
> tai
> 
> 336a = Musik (notation) tai 336a = Musik (performance) tai 336a = Tal

Mitä itse noita mäppäyksiä katselin, niin se 040$b vaikuttaa lopputulokseen, vaikka pakottaisi ajamaan ruotsinkielisellä. Joten varmaan parempi korjata nuo tietueet erämuokkauksella.

**Comment by AnttHeik** [comment]
@johannaraisa Nyt on taas päivitetty ruotsinkielisiä konversiosääntöjä, joten ajatko nuo TäTin testitietueet uudestaan?

**Comment by johannaraisa** [comment]
> @johannaraisa Nyt on taas päivitetty ruotsinkielisiä konversiosääntöjä, joten ajatko nuo TäTin testitietueet uudestaan?

Nyt on TäTissä, ja OUTIssa, meni rutiinilla sinne eka.

**Comment by AnttHeik** [comment]
> > @johannaraisa Nyt on taas päivitetty ruotsinkielisiä konversiosääntöjä, joten ajatko nuo TäTin testitietueet uudestaan?
> 
> Nyt on TäTissä, ja OUTIssa, meni rutiinilla sinne eka.

Jes, kiitokset. Hyvä että ajoit nuo OUTInkin, koska ovat vähän erilaisempia tietueita :)

**Comment by AnttHeik** [comment]
Nuo TäTin tietueet olivat jostain syystä jakautuneet nyt niin, että kolme meni ruotsinkielisellä ja ilmeisesti 9 suomenkielisillä, koska näyttävät niin oudoilta. Voisitko vielä ajaa kaikki 12 tietuetta uudestaan ruotsinkielisillä? :) Yleisesti ottaen näyttää muuten hyvältä nyt.

![kuva](https://github.com/user-attachments/assets/58c5f964-5578-4ddd-bab2-3c01ea996d44)

**Comment by johannaraisa** [comment]
> Nuo TäTin tietueet olivat jostain syystä jakautuneet nyt niin, että kolme meni ruotsinkielisellä ja ilmeisesti 9 suomenkielisillä, koska näyttävät niin oudoilta. Voisitko vielä ajaa kaikki 12 tietuetta uudestaan ruotsinkielisillä? :) Yleisesti ottaen näyttää muuten hyvältä nyt.
> 
> ![kuva](https://private-user-images.githubusercontent.com/116169663/359034586-58c5f964-5578-4ddd-bab2-3c01ea996d44.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQwNTY0MzcsIm5iZiI6MTcyNDA1NjEzNywicGF0aCI6Ii8xMTYxNjk2NjMvMzU5MDM0NTg2LTU4YzVmOTY0LTU1NzgtNGRkZC1iYWIyLTNjMDFlYTk5NmQ0NC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgxOVQwODI4NTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NTZlMDRhMTUxZGZmODAyMTc4ZmZiMDEwMzljNzQwYjY2NWE3OWUyNjA4ZTM0YTUwNTEyZjdlMjc1MjY0ZjQ0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.zC5sWPfKyvdL-bVxbfNU3Y8tIFosK8PB7xftYMxdgsU)

Se ruotsinkielisen määritelmähän meni, että halutaan vain tekstiaineisto ajettavan ruotsinkielisellä. Siksi tuo erittelu tulee noihin tiedostoihin.

**Comment by johannaraisa** [comment]
Tosin minusta nuo Kansalliskirjaston tekemät säännöt yrittää arvailla sen kielen, vaikka ajaisi suomeksikin. En siis ole varma kannattaako noita aineistoja yritää erotella.

**Comment by AnttHeik** [comment]
Aivan, niinhän se olikin. Mietin tuota kaiken aineiston ajamista suomenkielisillä säännöillä itsekin, mutta se vaatisi, että kaikissa vanhemmissakin tietueissa on muka 040b-kenttä, jossa arvo 'swe'. Lisäksi suomenkieliset konversiosäännöt eivät ihan täysin toimi, vaikka tuo 040b-kenttä olisikin mukana:
![kuva](https://github.com/user-attachments/assets/8781e4e2-b54f-48a6-b883-53ed51faf2b2)

**Comment by johannaraisa** [comment]
Yritetäänkö ajaa nuo kaikki ruotsinkielisillä ja katsotaan mille näyttää, vai mennäänkö tällä että vain tekstiaineistot?

**Comment by AnttHeik** [comment]
No voisihan ne ajaa, niin näkee osaako nuo ruotsinkieliset säännöt muitakin aineistoja kuin tekstiaineistot.

**Comment by johannaraisa** [comment]
Nyt on Tätissä uusi tiedosto, piti poistaa ne vanhat uuden tieltä.

**Comment by AnttHeik** [comment]
(Kommentoin tänne, kun kyse oli ruotsinkielisistä tietueista) Äh, ei se suomenkielinen konversio vieläkään osannut käsitellä tuota yhtä yllä olevaa tietuetta oikein, vaan osa termeistä kääntyy suomeksi, vaikka pitäisi pysyä ruotsina.

Aloin miettimään, että olisiko lopulta järkevintä vain ajaa ruotsinkielisellä konversiolla kaikki ne tietueet, joissa on 040b-kentän arvo swe eli jos muuttaisi tuota käsittelyehtoa alla olevalla tavalla. Menisiköhän se noin?
Yllä oleva ottaisi mukaan ne vanhat tekstiaineistot, joissa ei 040b-kenttää ole, ja alla oleva ehto takaisi sen, että kaikki uudemmat aineistot, joissa on kuvailukieleksi merkitty swe, menisivät ruotsinkielisellä konversiolla (myös nykyiset RDA-tietueet).

000/06 = a
008/35-37 = swe
041a = swe
tai 
040b = swe

**Comment by AnneliO** [comment]
RDA-konversiot on tehty, onko tämä tiketti vielä ajankohtainen vai voisiko sen sulkea?

**Comment by AnttHeik** [comment]
Voi sulkea.
